pkg/apis/workloads/v1alpha1/console_types.go

```diff
@@ -131,3 +162,38 @@ type ConsoleTemplateList struct {
 
 	Items []ConsoleTemplate `json:"items"`
 }
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ConsoleAuthorisation declares a console authorisation that is instantiated when a console that requires authoristion is created. It is used to store authorisations for the associated console.
+type ConsoleAuthorisation struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec ConsoleAuthorisationSpec `json:"spec"`
+}
+
+// ConsoleAuthorisationSpec defines the specification for a console authorisation
+type ConsoleAuthorisationSpec struct {
+	// The reference to the console by name that this ConsoleAuthorisation belongs to.
+	ConsoleRef struct {
```

Comment:
Should we not be using `corev1.LocalObjectReference` here?

See example earlier in the file

Reply:
👍